src/datasources/exchange-api/exchange-api.service.spec.ts

```diff
@@ -37,23 +44,98 @@ describe('ExchangeApi', () => {
         USD: 'Dollar',
       })
       .build();
-    mockNetworkService.get.mockResolvedValue({ data: expectedFiatCodes });
+    mockCacheFirstDataSource.get.mockResolvedValue(expectedFiatCodes);
 
     const fiatCodes = await service.getFiatCodes();
 
     expect(fiatCodes).toBe(expectedFiatCodes);
   });
 
+  it('fiatCodes uses default cache TTL (12h) if none is set', async () => {
+    const expectedFiatCodes = exchangeFiatCodesBuilder().build();
+    mockCacheFirstDataSource.get.mockResolvedValue(expectedFiatCodes);
+
+    await service.getFiatCodes();
+
+    expect(mockCacheFirstDataSource.get).toBeCalledWith(
+      new CacheDir('exchange_fiat_codes', ''),
+      `${exchangeBaseUri}/symbols`,
+      { params: { access_key: exchangeApiKey } },
+      60 * 60 * 12, // 12h in seconds
+    );
+  });
+
+  it('fiatCodes uses set cache TTL', async () => {
+    const exchangeBaseUri = faker.internet.url();
+    const exchangeApiKey = faker.random.alphaNumeric();
+    const ttl = 60;
+    const fakeConfigurationService = new FakeConfigurationService();
+    fakeConfigurationService.set('exchange.baseUri', exchangeBaseUri);
+    fakeConfigurationService.set('exchange.apiKey', exchangeApiKey);
+    fakeConfigurationService.set('exchange.cacheTtlSeconds', ttl);
+    const expectedFiatCodes = exchangeFiatCodesBuilder().build();
+    mockCacheFirstDataSource.get.mockResolvedValue(expectedFiatCodes);
+    const target = new ExchangeApi(
+      fakeConfigurationService,
+      mockCacheFirstDataSource,
+    );
+
+    await target.getFiatCodes();
+
+    expect(mockCacheFirstDataSource.get).toBeCalledWith(
+      new CacheDir('exchange_fiat_codes', ''),
+      `${exchangeBaseUri}/symbols`,
+      { params: { access_key: exchangeApiKey } },
+      ttl, // 60 seconds
+    );
+  });
```

Comment:
The way this test is setup makes it less clear on why I'm creating a new instance of `ExchangeApi` here. The reason is because we set environment variables that are read in its constructor. Changing the configuration just for the test wouldn't work without creating a new instance.

I'm more in favour of removing the `beforeEach` in this case and do a setup on each test (the test `should error if configuration is not defined` was already doing something similar so I decided not to change the pattern right now).